File: android/app/src/main/kotlin/com/hyequest/audiorecorder/data/audiorecorder/RecordStatusModel.kt
package com.hyequest.audiorecorder.data.audiorecorder

data class RecordStatusModel(
    val isRecording: Boolean,
    val recordDuration: Double
) {
    fun toMap(): Map<String, Any> {
        return mapOf(
            "isRecording" to isRecording,
            "recordDuration" to recordDuration
        )
    }
}

File: ios/Runner/data/AudioRecorder/RecordStatusModel.swift
//
//  RecordStatusModel.swift
//  Runner
//
//  Created by Emmanuel Idaresit on 10/08/2025.
//
import Foundation

class RecordStatusModel {
    let isRecording: Bool
    let recordDuration: TimeInterval
    
    init(isRecording: Bool, recordDuration: TimeInterval) {
        self.isRecording = isRecording
        self.recordDuration = recordDuration
    }
    
    func toDictionary() -> [String: Any] {
        return [
            "isRecording": isRecording,
            "recordDuration": recordDuration,
        ]
    }
}
